### PR TITLE
Ensure case-insensitive comparisons against overridden file extension values (plus additional tests to support case-insensitive checks)

### PR DESCRIPTION
**Pull request by elohanlon**
Also bundled some general rubocop updates into this PR (and now referencing our rubocul gem).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby-version: ['2.5.0', '2.6.4', '2.7.5', '3.0.3']
+        ruby-version: ['2.7.5', '3.0.3', '3.1.4', '3.2.2']
```
Comment (barmintor) [inline .github/workflows/ci.yml:12]: 🎉

File: spec/unit/lib/best_type/dc_type_lookup_spec.rb [hunk 2]
```diff
@@ -36,6 +36,12 @@
       expect(dc_type_lookup.instance_variable_get(:@mime_type_lookup)).to receive(:for_file_name).with(file_name).and_call_original
       dc_type_lookup.for_file_name(file_name)
     end
+
+    ['file.jpg', 'file.Jpg', 'file.JPG'].each do |variant|
```
Comment (barmintor) [inline spec/unit/lib/best_type/dc_type_lookup_spec.rb:40]: 👍

**Review by barmintor — APPROVED** [review]
Seems like it will solve some problems!

**Comment by elohanlon** [comment]
Thank you for the review!

**Pull request merged**